From ccdfd293f9e948f0f62ac4e9924d72539a4e81ee Mon Sep 17 00:00:00 2001
From: Mantas Pucka <mantas@8devices.com>
Date: Mon, 2 Jun 2025 17:19:45 +0300
Subject: [PATCH] net: pcs: ipq-uniphy: keep autoneg enabled in SGMII mode

For PHYs that don't use in-band-status (e.g. 2.5G PHY swiching between
SGMII and 2500base-x), SGMII autoneg still must be enabled. Only mode
that should use forced speed is 1000base-x

Signed-off-by: Mantas Pucka <mantas@8devices.com>
---
 drivers/net/pcs/pcs-qcom-ipq9574.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/pcs/pcs-qcom-ipq9574.c
+++ b/drivers/net/pcs/pcs-qcom-ipq9574.c
@@ -431,7 +431,7 @@ static int ipq_pcs_config_sgmii(struct i
 	/* Nothing to do here as in-band autoneg mode is enabled
 	 * by default for each PCS MII port.
 	 */
-	if (neg_mode == PHYLINK_PCS_NEG_INBAND_ENABLED)
+	if (interface != PHY_INTERFACE_MODE_1000BASEX)
 		return 0;
 
 	/* Set force speed mode */
